server/data/data-hbase/src/main/java/org/apache/james/rrt/hbase/HBaseRecipientRewriteTable.java

```diff
@@ -225,10 +219,9 @@ private void doUpdateMapping(String user, Domain domain, String mapping) throws
      * 
      * @param user the user
      * @param domain the domain
-     * @param mapping the mapping
      * @throws RecipientRewriteTableException
      */
-    private void doRemoveMapping(String user, Domain domain, String mapping) throws RecipientRewriteTableException {
+    private void doRemoveMapping(String user, Domain domain) throws RecipientRewriteTableException {
```

Comment:
this one is there strange, but let's keep it like this for now. It means that hbase impl is buggy, no?

Reply:
No it's not. HBase explicitly call `doUpdateMapping` when more than one mapping exist, thus making it non buggy: doRemove is only called upon single mapping.

(Of course until you parrallel update it: then you break it....)